src/api/v2/v2AzureResourcesApi.ts

```diff
@@ -179,6 +179,10 @@ export interface ResourcePickOptions {
  * The current (v2) Azure Resources extension API.
  */
 export interface V2AzureResourcesApi extends AzureResourcesApiBase {
+
+    applicationResourceTreeDataProvider: vscode.TreeDataProvider<unknown>;
```

Comment:
For the purposes of the API, these can probably be `vscode.TreeDataProvider<ResourceGroupsItem>` or something along those lines.

Reply:
Ok. I wasn't sure how much we wanted to hide from the clients.

Reply:
I think they'll almost _need_ at least something, e.g. to indicate that the objects are wrappers. What did we say as the input to the quick pick experience? Isn't that `ResourceGroupsItem`?

Reply:
Currently the quick pick experience needs a `TreeDataProvider<unknown>`